src/components/user-list.js

```diff
@@ -15,51 +18,82 @@ export default class UserList extends React.Component {
 
     _getAppState () {
         return {
-            duplicateUserId: '',
-            wantAdd:         false,
-            users:           UserStore.getUsers()
+            duplicateUserId:   '',
+            highlightedFields: '',
+            addUserMode:       false,
+            searchMode:        false,
+            users:             UserStore.getUsers()
```

Comment:
Honestly it's not a common way to reset all state fields in `_getAppState` method that used in `change` handler.
Let's at least rename `_getAppState ` -> `_getAppInitialState `